okonomiyaki/file_formats/_egg_info.py

```diff
@@ -573,14 +572,25 @@ def to_string(self):
         template = _METADATA_TEMPLATES.get(self.metadata_version, None)
         data = self._to_dict()
 
+        if six.PY2:
+            # Hack to avoid the 'u' prefix to appear in the spec/depend entries
+            for k, v in data.items():
+                data[k] = encode_if_needed(v)
+
         # This is just to ensure the exact same string as the produced by the
         # legacy buildsystem
         if len(self.packages) == 0:
             data[_TAG_PACKAGES] = "[]"
         else:
-            data[_TAG_PACKAGES] = "[\n{0}\n]". \
-                format("\n".join("  '{0}',".format(p)
-                       for p in self.packages))
+            if six.PY2:
+                packages = [decode_if_needed(p) for p in self.packages]
+            else:
+                packages = self.packages
+            data[_TAG_PACKAGES] = (
+                "[\n{0}\n]".format(
```

Comment:
I think this will be unicode on py3 and a byte string in py2, is that intended?`